internal/server/handlers.go

```diff
@@ -58,33 +58,67 @@ func (a *API) CreateIdentity(c *gin.Context, r *api.CreateIdentityRequest) (*api
 		Kind: kind,
 	}
 
-	if err := access.CreateIdentity(c, identity); err != nil {
-		return nil, err
-	}
+	// identity creation linked to a provider should be attempted even if an identity is already known
+	if r.ProviderName != "" {
```

Comment:
I don't think CreateIdentity should be concerned with providers. Identity is a stand-alone infra-owned thing now